aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -558,9 +558,10 @@ fn load_constructor_function(
     function_name: &IdentStr,
     expected_return_ty: &Type,
 ) -> VMResult<LoadedFunction> {
-    // Here, we do not charge gas for module loading due to invoking a constructor function. This
-    // is safe to do because all constructor functions are located at 0x1 (special address) and so
-    // should not be charged.
+    // MODULE METERING SAFETY:
```

Comment:
What is a constructor function? Is this related to parsing arguments (like Object) for transactions?

If so, then this assumption will soon be wrong, as we plan to introduce public structs/enums. The current idea to implement those is by special functions (so if `m::C` is a constructor, there will be a `public mk$C` function in `m`). We can ignore this for now, however, I wonder whether making metering here available would complicate things.

Reply:
Yes, changed the comment to make this clear. 

Well, I am propagating gas meter here in the PR one after the next one, so having metering here is possible. For now, I added a check here to make sure if we add public constructors - this catches the invariant is violated.

Reply:
And we probably redesign constructor functions anyway, so probably we can revisit this once public structs have some draft implementation.